pkg/aws/services/vpclattice.go

```diff
@@ -313,60 +318,78 @@ func (d *defaultLattice) ListServiceNetworkServiceAssociationsAsList(ctx context
 	return result, nil
 }
 
-func (d *defaultLattice) FindServiceNetwork(ctx context.Context, name string, optionalAccountId string) (*ServiceNetworkInfo, error) {
+func (d *defaultLattice) snSummaryToLog(snSum []*vpclattice.ServiceNetworkSummary) []string {
+	out := make([]string, len(snSum))
+	for i, s := range snSum {
+		out[i] = fmt.Sprintf("name=%s, id=%s", aws.StringValue(s.Name), aws.StringValue(s.Id))
+	}
+	return out
+}
+
+func (d *defaultLattice) FindServiceNetwork(ctx context.Context, nameOrId string) (*ServiceNetworkInfo, error) {
 	// When default service network is provided, override for any kind of SN search
 	if config.ServiceNetworkOverrideMode {
-		name = config.DefaultServiceNetwork
+		nameOrId = config.DefaultServiceNetwork
 	}
-	input := vpclattice.ListServiceNetworksInput{}
-
-	var innerErr error
-	var snMatch *ServiceNetworkInfo
-	err := d.ListServiceNetworksPagesWithContext(ctx, &input, func(page *vpclattice.ListServiceNetworksOutput, lastPage bool) bool {
-		for _, r := range page.Items {
-			if aws.StringValue(r.Name) != name {
-				continue
-			}
-			acctIdMatches, err1 := accountIdMatches(optionalAccountId, *r.Arn)
-			if err1 != nil {
-				innerErr = err1
-				return false
-			}
-			if !acctIdMatches {
-				continue
-			}
-
-			tagsInput := vpclattice.ListTagsForResourceInput{
-				ResourceArn: r.Arn,
-			}
 
-			tagsOutput, err2 := d.ListTagsForResourceWithContext(ctx, &tagsInput)
-			if err2 != nil {
-				innerErr = err2
-				return false
-			}
-
-			snMatch = &ServiceNetworkInfo{
-				SvcNetwork: *r,
-				Tags:       tagsOutput.Tags,
-			}
-			return false
-		}
+	input := &vpclattice.ListServiceNetworksInput{}
+	allSn, err := d.ListServiceNetworksAsList(ctx, input)
+	if err != nil {
+		return nil, err
+	}
 
-		return true
+	// Try find by name first, if there is no single match, continue with id
+	// match. Ideally name match should work just fine, but in desperate
+	// scenario of shared SN when naming collision happens using ID can be
+	// an option
+	var snMatch *vpclattice.ServiceNetworkSummary
+	nameMatch := utils.SliceFilter(allSn, func(snSum *vpclattice.ServiceNetworkSummary) bool {
+		return aws.StringValue(snSum.Name) == nameOrId
+	})
+	idMatch := utils.SliceFilter(allSn, func(snSum *vpclattice.ServiceNetworkSummary) bool {
+		return aws.StringValue(snSum.Id) == nameOrId
 	})
```

Comment:
For the discussion in slack, filter by name then id make sense. We could also add another filter by ARN, i.e. change function name to 
`FindServiceNetwork(ctx context.Context, nameOrIdOrArn string)` and add a `arnMatch=`

Reply:
I would drop ID and use ARN only

Reply:
ARN is not a valid name for Gateway, "/" if not allowed